src/planner/ompl/Planner.cpp

```diff
@@ -317,6 +317,68 @@ trajectory::InterpolatedPtr planCRRTConnect(
                   std::move(_interpolator), _maxPlanTime);
 }
 
+//=============================================================================
+trajectory::InterpolatedPtr simplifyOMPL(statespace::StateSpacePtr _stateSpace,
+                                           statespace::InterpolatorPtr _interpolator,
+                                           distance::DistanceMetricPtr _dmetric,
+                                           constraint::SampleablePtr _sampler,
+                                           constraint::TestablePtr _validityConstraint,
+                                           constraint::TestablePtr _boundsConstraint,
+                                           constraint::ProjectablePtr _boundsProjector, 
+                                           double _maxDistanceBtwValidityChecks,
+                                           trajectory::InterpolatedPtr _originalTraj)
+{
+
+// Step 1: Generate the space information of OMPL type
+auto si = getSpaceInformation(
+      _stateSpace, _interpolator, std::move(_dmetric), std::move(_sampler),
+      std::move(_validityConstraint), std::move(_boundsConstraint),
+      std::move(_boundsProjector), _maxDistanceBtwValidityChecks);
+
+// Get the state space
+auto sspace = ompl_static_pointer_cast<GeometricStateSpace>(
+      si->getStateSpace());
+
+
+// Step 2: Convert the AIKIDO trajectory to Geometric Path 
+::ompl::geometric::PathGeometric *path = new ::ompl::geometric::PathGeometric(si);
+
+for (size_t idx = 0; idx < _originalTraj->getNumWaypoints(); ++idx) 
+{
+      auto ompl_state = sspace->allocState(_originalTraj->getWaypoint(idx));
+      path->append(ompl_state);
+      sspace->freeState(ompl_state);
+}
+
+// Step 3: Use the OMPL methods to simplify the path
+::ompl::geometric::PathSimplifierPtr simplifier;
+simplifier.reset(new ::ompl::geometric::PathSimplifier(si));
+bool const shortened = simplifier->shortcutPath(*path, 1, 1, 1.0, 0.005); 
```

Comment:
Make `maxSteps`, `maxEmptySteps`, `rangeRatio`, and `snapToVertex` parameters of this function.

Also, note that the current hard-coded settings are (in my opinion) nonsensical. Setting `maxSteps = 1` means that it will only try to shortcut two randomly selected points before returning. The default values in OMPL are a more reasonable choice, possibly with the modification that `rangeRatio = 1.0` and `snapToVertex = 0.0`.

Another option is to leave `maxSteps = 1`, but call `shortcutPath` in a loop until a time limit (or other termination condition) expires. This is the approach I used for `OMPLSimplifier` in `or_ompl`.